csharp/src/Ice/Logger.cs

```diff
@@ -8,161 +8,123 @@
 
 namespace ZeroC.Ice
 {
-    public abstract class Logger : ILogger
+    /// <summary>Represents a logger that writes messages to the standard error output stream.</summary>
+    public sealed class ConsoleLogger : Logger
     {
-        public Logger(string prefix)
-        {
-            Prefix = prefix;
-
-            if (prefix.Length > 0)
-            {
-                FormattedPrefix = prefix + ": ";
-            }
-            else
-            {
-                FormattedPrefix = "";
-            }
-        }
+        private static readonly object _globalMutex = new object();
 
-        public void Print(string message)
+        /// <summary>Creates a new console logger.</summary>
+        /// <param name="prefix">The prefix to perpend to messages write by this logger.</param>
+        public ConsoleLogger(string prefix)
+            : base(prefix)
         {
-            lock (GlobalMutex)
-            {
-                Write(message);
-            }
         }
 
-        public virtual void Trace(string category, string message)
+        public override ILogger CloneWithPrefix(string prefix) => new ConsoleLogger(prefix);
+
+        protected override void Write(string message)
         {
-            string s = Format("--", category, message);
-            lock (GlobalMutex)
+            lock (_globalMutex)
```

Comment:
Is this locking necessary? System.Console.Error.WriteLine must be thread-safe.

Reply:
Right seems we can get rid of it, according to https://docs.microsoft.com/en-us/dotnet/api/system.console?view=netcore-3.1#console-io-streams